src/app.ts

```diff
@@ -67,10 +69,12 @@ function withSpidAuth(
     passport.authenticate("spid", async (err, user) => {
       if (err) {
         log.error("Error in SPID authentication: %s", err);
-        const errorCode = getErrorCodeFromResponse(err.statusXml);
         return res.redirect(
           clientErrorRedirectionUrl +
-            (errorCode ? `?errorCode=${errorCode}` : "")
+            fromNullable(err.statusXml)
+              .chain(statusXml => getErrorCodeFromResponse(statusXml))
+              .chain(errorCode => some(`?errorCode=${errorCode}`))
```

Comment:
this should be `.map(errorCode =>`?errorCode=${errorCode}`)`

Reply:
fixed